wcivf/apps/elections/tests/test_election_views.py

```diff
@@ -141,6 +141,107 @@ def test_num_candidates(self):
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "elections/post_view.html")
+        self.assertTemplateUsed(
+            response, "elections/includes/_post_meta_title.html"
+        )
+        self.assertTemplateUsed(
+            response, "elections/includes/_post_meta_description.html"
+        )
+        self.assertContains(response, "This election has been cancelled.")
+        self.assertNotContains(response, "No candidates known yet.")
+
+    # In the case of fewer candidates than seats but where number of candidates is one or more
+    def test_uncontested_and_rescheduled(self):
+        self.post_election.winner_count = 5
+        people = [PersonFactory() for p in range(4)]
+        for person in people:
+            PersonPostFactory(
+                post_election=self.post_election,
+                election=self.election,
+                post=self.post,
+                person=person,
+            )
+        self.post_election.election.uncontested = True
+        self.post_election.election.save()
+
+        response = self.client.get(
+            self.post_election.get_absolute_url(), follow=True
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "elections/post_view.html")
+
+        self.assertContains(
+            response,
+            "This election was uncontested because the number of candidates who stood was fewer than the minimum required.",
+        )
+        self.assertNotContains(response, "This election was cancelled.")
+
+    # In the case where candidates and seats are equal
+    def test_uncontested_and_elected(self):
+        self.post_election.winner_count = 4
+        people = [PersonFactory() for p in range(4)]
+        for person in people:
+            PersonPostFactory(
+                post_election=self.post_election,
+                election=self.election,
+                post=self.post,
+                person=person,
+            )
+        self.post_election.election.uncontested = True
+        self.post_election.election.save()
+        response = self.client.get(
+            self.post_election.get_absolute_url(), follow=True
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "elections/post_view.html")
+
+        self.assertContains(
+            response,
+            "This election was uncontested because the number of candidates who stood was equal to the number of available seats.",
+        )
+        self.assertContains(
+            response,
+            "No votes will be cast, and the candidate below has been automatically declared the winner.",
+        )
+        self.assertNotContains(response, "This election was cancelled.")
+
+    # In the case where the number of candidates is equal to seats but then the candidate dies
+    def test_cancelled_candidate_died(self):
```

Comment:
@symroe I'm still not clear on this. Do we want a specific message in this case?